R/derive_seq.R

```diff
@@ -0,0 +1,67 @@
+#' Derive the sequence number (`--SEQ`) variable
+#'
+#' @description
+#' [derive_seq()] creates a new identifier variable: the sequence number
+#' (`--SEQ`).
+#'
+#' This function adds a newly derived variable to `tgt_dat`, namely the sequence
+#' number (`--SEQ`) whose name is the one provided in `tgt_var`. An integer
+#' sequence is generated that uniquely identifies each record within the domain.
+#'
+#' Prior to the derivation of `tgt_var`, the data frame `tgt_dat` is sorted
+#' according to grouping variables indicated in `rec_vars`.
+#'
+#' @param tgt_dat The target dataset, a data frame.
+#' @param tgt_var The target SDTM variable: a single string indicating the name
+#'   of the sequence number (`--SEQ`) variable, e.g. `"DSSEQ"`. Note that
+#'   supplying a name not ending in `"SEQ"` will raise a warning.
+#' @param rec_vars A character vector of record-level identifier variables. Use
+#' [rec_vars()] to quickly get the identifier variables for a given domain.
```

Comment:
Does rec_vars() pull the default values from "data-raw/domain_record_vars.csv.xz"? Does this file come from {roak}? If so, I will make it a CSV and make it viewable to the users. There is no guarantee that the keys mentioned in the file are accurate.

Reply:
Also, I think we need to document this as sample key variables.

Reply:
- `rec_vars()` pulls the default values from an internal data set that becames bundled with the package. The source for this data set is explained here: https://github.com/pharmaverse/sdtm.oak/pull/53/commits/8662fa713175a3eeeeb6dcfc64e82093e5213dc8#diff-d23bac3cea2c5fe24e84e8c626ec809f57783e96bfa6e7b606eda309dfadc994R1-R4.
- Instead of making the CSV available to users, an exported data set would be more appropriate. I can change that internal data set to an exported data set.
- If those default keys are not appropriate for the user specific case, they can always pass it as there is an argument `rec_vars()`.
- Regarding the naming of those key variables as "samples" key variables, I tried to follow CDISC SDTM terminology, and to my understanding those unique rows to be identified as referred to as records in CDISC terminology. So I will leave it be and let you confirm.